R/aggregate_game_stats_def.R

```diff
@@ -0,0 +1,524 @@
+    ) %>%
+    dplyr::summarise(recovery_yards = sum(.data$fumble_recovery_1_yards)) %>%
+    dplyr::filter(!is.na(.data$player_id)) %>%
+    dplyr::bind_rows(
+      fumble_yds_opp_data %>%
+        dplyr::group_by(
+          .data$season, .data$week,
+          "player_id" = .data$fumble_recovery_2_player_id
+        ) %>%
+        dplyr::summarise(recovery_yards = sum(.data$fumble_recovery_2_yards)) %>%
+        dplyr::filter(!is.na(.data$player_id))
+    ) %>%
+    dplyr::group_by(.data$season, .data$week, .data$player_id) %>%
+    dplyr::summarise(fumble_recovery_yards_opp = sum(.data$recovery_yards)) %>%
+    dplyr::ungroup()
+
+  # Penalty stats -----------------------------------------------------------
+
+  # get penalty stats
+  penalty_df <- penalty_data %>%
+    dplyr::filter(
+      !is.na(.data$penalty_player_id),
+      .data$defteam == .data$penalty_team
+    ) %>%
+    dplyr::select(
+      "season", "week", "yards_gained",
+      tidyselect::contains("penalty_player_") & tidyselect::ends_with("_id")
+    ) %>%
+    tidyr::pivot_longer(
+      cols = c(
+        tidyselect::contains("penalty_player_")
+      ),
+      names_to = "desc",
+      names_prefix = "penalty_",
+      values_to = "penalty_player_id",
+      values_drop_na = TRUE
+    ) %>%
+    dplyr::mutate(n = 1) %>%
+    tidyr::pivot_wider(
+      names_from = desc,
+      values_from = n,
+      values_fn = sum,
+      values_fill = 0
+    ) %>%
+    dplyr::select(
+      "season", "week",
+      "player_id" = "penalty_player_id",
+      "penalty" = "player_id"
+    )
+
+  # Touchdown stats -----------------------------------------------------------
+
+  # get defensive touchdowns
+  touchdown_df <- data %>%
+    dplyr::filter(.data$touchdown == 1) %>%
+    dplyr::filter(.data$defteam == .data$td_team) %>%
+    dplyr::group_by(
+      .data$season, .data$week,
+      "player_id" = .data$td_player_id
+    ) %>%
+    dplyr::summarise(td = sum(.data$touchdown)) %>%
+    dplyr::ungroup()
+
+  # Combine all stats -------------------------------------------------------
+
+  # combine all the stats together
+
+  player_df <- tackle_df %>%
+    dplyr::full_join(tackle_yards_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(pressure_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(int_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(safety_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(fumble_df_own, by = c("season","week","player_id")) %>%
+    dplyr::full_join(fumble_df_opp, by = c("season","week","player_id")) %>%
+    dplyr::full_join(fumble_yds_own_df, by = c("season","week","player_id")) %>%
+    dplyr::full_join(fumble_yds_opp_df, by = c("season","week","player_id")) %>%
+    dplyr::full_join(penalty_df, by = c("season","week","player_id")) %>%
+    dplyr::full_join(touchdown_df, by = c("season","week","player_id")) %>%
```

Comment:
I was lazy and did not add team to these data frames. This needs to be done because some players wouldn't show in a team if they are not in the data frame above

Reply:
agree and think that team should be carried forward from data for all of offense, defense, and kicking (rather than being brought in via rosters/recent_team etc).

Reply:
Team added to all intermediate data frames in https://github.com/nflverse/nflfastR/pull/288/commits/bae2bd867e90eba88c9af731b54d0cc726f75561